pkg/ingester/ingester.go

```diff
@@ -667,6 +676,13 @@ type pushStats struct {
 	perMetricSeriesLimitCount int
 }
 
+type pushStats struct {
+	succeededExemplarsCount int
+	failedExemplarsCount    int
+	floatStats              typedPushStats
+	histogramStats          typedPushStats
```

Comment:
Given we no longer track float vs histogram metrics anymore, do we need to complicate the code with `typedPushStats`?

Reply:
Came up with a compromise: keep existing stats and count both floats and histograms towards samples, but add  two new stats for successful and failed histograms that only take into account histograms. This way we can have debug logs and statistics without complicating the code and changing the meaning of metrics.